File: Day_002/moving-all-zeros-to-end.py
class Solution:
    def pushZerosToEnd(self, arr):
        c = 0
         
        for num in arr:
            if num != 0:
                arr[c] = num
                c += 1
        
        while c < len(arr) :
            arr[c] = 0
            c += 1
    	    
        # Alternative approach :
        # for num in arr :
        #     if num != 0 :
        #         num, arr[c] = arr[c], num
        #         c += 1

File: moving-all-zeros-to-end.java
class Solution {
    void pushZerosToEnd(int[] arr) {
        int temp, c = 0;
        
        for (int i = 0; i < arr.length; i++){
            if (arr[i] != 0){
                temp = arr[i];
                arr[i] = arr[c];
                arr[c] = temp;
                c++;
            }
        }
    }
}